changelog/pending/20240621--engine--fix-provider-delete-s-broken-by-16302.yaml

```
@@ -0,0 +1,4 @@
+changes:
+- type: fix
+  scope: engine
+  description: "Fix provider `Delete`s broken by #16302"
```

```suggestion
  description: "Fix provider `Delete`s"
```

Not sure if we need to include the original PR here, but I'm fine with it either way.